src/main/java/fr/esgi/al5_2/Tayarim/services/AuthService.java

```diff
@@ -3,87 +3,135 @@
 import lombok.NonNull;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.AbstractMap;
+import java.util.Map.Entry;
 import java.util.UUID;
 
 @Service
 @Transactional(readOnly = true)
 public class AuthService {
 
     private final ProprietaireService proprietaireService;
+    private final AdministrateurService administrateurService;
     private final JwtHelper jwtHelper;
 
     private final TokenCacheService tokenCacheService;
 
-    public AuthService(ProprietaireService proprietaireService, JwtHelper jwtHelper, TokenCacheService tokenCacheService) {
+    public AuthService(ProprietaireService proprietaireService, AdministrateurService administrateurService, JwtHelper jwtHelper, TokenCacheService tokenCacheService) {
         this.proprietaireService = proprietaireService;
+        this.administrateurService = administrateurService;
         this.jwtHelper = jwtHelper;
         this.tokenCacheService = tokenCacheService;
     }
 
-    public AuthLoginResponseDTO loginProprietaire(@NonNull String email, @NonNull String password){
-        ProprietaireDTO proprietaireDTO = proprietaireService.getProprietaireByEmail(email);
+    @Transactional(readOnly = true, propagation = Propagation.NOT_SUPPORTED)
+    public AuthLoginResponseDTO login(@NonNull String email, @NonNull String password) {
+        ProprietaireDTO proprietaireDTO = null;
+        AdministrateurDTO administrateurDTO = null;
+        Long id;
+        boolean isAdmin = false;
+        try {
+            proprietaireDTO = proprietaireService.getProprietaireByEmail(email);
+            id = proprietaireDTO.getId();
+        } catch (Exception e) {
+            try {
+                administrateurDTO = administrateurService.getAdministrateurByEmail(email);
+                id = administrateurDTO.getId();
+                isAdmin = true;
+            } catch (Exception exception) {
+                throw new UtilisateurNotFoundException();
+            }
 
-        if(!proprietaireService.verifyPassword(password, proprietaireDTO.getId())){
+        }
+
+        if (proprietaireDTO != null && !proprietaireService.verifyPassword(password, proprietaireDTO.getId())) {
             throw new ProprietaireNotFoundException();
+        } else if (administrateurDTO != null && !administrateurService.verifyPassword(password, administrateurDTO.getId())) {
+            throw new AdministrateurNotFoundException();
         }
 
-        String uuid = tokenCacheService.getFromCache(proprietaireDTO.getId());
-        if (uuid == null){
+        String uuid = tokenCacheService.getFromCache(id);
+        if (uuid == null) {
             uuid = UUID.randomUUID().toString();
-            tokenCacheService.addToCache(proprietaireDTO.getId() ,uuid);
+            tokenCacheService.addToCache(id, uuid);
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' child has incorrect indentation level 12, expected level should be 6.